lms/static/js/Markdown.Editor.js

```diff
@@ -1040,107 +1057,144 @@
                 // doesn't change url if started with '/' (local)
-                if (!/^(?:https?|ftp):\/\//.test(text) && text.charAt(0) != '/') {
-                    text = 'http://' + text;
+                if (!/^(?:https?|ftp):\/\//.test(url) && url.charAt(0) !== '/') {
+                    url = 'http://' + url;
                 }
             }
+            var isValidUrl = /^((?:http|https|ftp):\/{2}|\/)[^]+$/.test(url),
+                isValidDesc = (
+                    descInput.checkValidity() &&
+                    (descInput.required ? description.length : true)
+                );
+
+            if ((isValidUrl && isValidDesc) || isCancel) {
+                dialog.parentNode.removeChild(dialog);
+                callback(url, description);
+            } else {
+                var error_count = 0;
+                if (!isValidUrl) {
+                    urlInput.classList.add('has-error');
+                    urlErrorMsg.style.display = 'inline-block';
+                    error_count += 1;
+                } if (!isValidDesc) {
+                    descInput.classList.add('has-error');
+                    descErrorMsg.style.display = 'inline-block';
+                    error_count += 1;
+                }
 
-            dialog.parentNode.removeChild(dialog);
+                document.getElementById('wmd-editor-dialog-form-errors').textContent = [
+                    error_count,
+                    error_count === 1 ? gettext('error found in form.') : gettext('errors found in form.'),
+                    !isValidUrl ? urlErrorMsg.textContent : '',
+                    !isValidDesc ? descErrorMsg.textContent : ''
+                ].join(' ');
+
+                document.getElementById('wmd-editor-dialog-form-errors').focus();
+            }
 
-            callback(text);
             return false;
         };
 
-
-
         // Create the text input box form/window.
         var createDialog = function () {
+            var fileUloadButtonHtml = imageUploadHandler ? (
+                    '<button id="file-upload-proxy" class="btn btn-primary btn-base form-btn">' +
+                        gettext("Choose File") +
+                    '</button>' +
+                    '<input type="file" name="file-upload" id="file-upload" style="display:none;"/>') : '',
+
+                imageIsDecorativeInputHtml = imageUploadHandler ? (
+                        '<label for="img-is-descriptive" class="field-label label-inline">' +
+                            '<input type="checkbox" id="img-is-descriptive" class="field-input input-checkbox">' +
+                            '<span class="field-input-label"> ' + imageIsDecorativeLabel + '</span>' +
+                        '</label>'
+                    ) : '',
+
+                descError = (imageUploadHandler ?
+                    gettext("Please describe this image or agree that it has no contextual " +

```

Comment:
It would be a good idea to generate i18n strings on your sandbox and then verify that all the strings are translating correctly. See this document for details:

http://edx-developer-guide.readthedocs.org/en/latest/internationalization/i18n.html#coverage-testing